packages/exchange-analytics/test/integration/adapter.test.ts

```diff
@@ -0,0 +1,46 @@
+import http from 'http'
+import type { AddressInfo } from 'net'
+import process from 'process'
+import request from 'supertest'
+import type { SuperTest, Test } from 'supertest'
+
+import { server as startServer } from '../../src'
+import { testFundingRates } from './funding-rates/get'
+
+let oldEnv: NodeJS.ProcessEnv
+
+export interface SuiteContext {
+  server: http.Server
+  req: SuperTest<Test>
+}
+
+beforeAll(() => {
+  oldEnv = JSON.parse(JSON.stringify(process.env))
+  process.env.CACHE_ENABLED = 'false'
+  process.env.WARMUP_ENABLED = 'false'
+  process.env.API_KEY = 'fake-api-key'
+})
+
+afterAll(() => {
+  process.env = oldEnv
+})
+
+describe('execute', () => {
+  const context: SuiteContext = {
+    server: null,
+    req: null,
+  }
+
+  beforeEach(async () => {
+    context.server = await startServer()
+    context.req = request(`localhost:${(context.server.address() as AddressInfo).port}`)
+  })
+
+  afterEach((done) => {
+    context.server.close(done)
+  })
+
+  describe('exchange-analytics', () => {
```

Comment:
We are testing endpoints here. `exchange-analytics` is  the EA name we already now
```typescript
describe('endpoint', () => {
```